### ISSUE TITLE
multi-GPU Trainning

### ISSUE THREAD

**Issue by Maplessaq** [Maplessaq/1#2]
During parameter ablation studies, I attempted to set batch size to 32 but encountered Out-Of-Memory (OOM) errors. To resolve this, I implemented a dual-GPU solution using two NVIDIA RTX 3090 24GB graphics cards, successfully achieving batch size 32 for ablation experiments.

import torch
import torch.nn as nn

# Initialize dual-GPU environment
device_ids = [0, 1]  # Two GPUs (e.g., RTX 3090)
model = YourModel()
model = nn.DataParallel(model, device_ids=device_ids)
model.to(f'cuda:{device_ids[0]}')

# Training loop
for inputs, labels in dataloader:
    inputs = inputs.to(f'cuda:{device_ids[0]}')
    outputs = model(inputs)  # Automatic distributed processing
    loss = criterion(outputs, labels)
    loss.backward()
    optimizer.step()
